Sources/Amplitude/Utilities/AccessibilityHierarchyParser.swift

```diff
@@ -0,0 +1,217 @@
+import Accessibility
+import UIKit
+
+public struct AccessibilityTarget {
+
+    // MARK: - Public Types
+
+    public enum Shape {
+
+        /// Accessibility frame, in the coordinate space of the view being processed.
+        case frame(CGRect)
+
+        /// Accessibility path, in the coordinate space of the view being processed.
+        case path(UIBezierPath)
+
+        public func contains(_ point: CGPoint) -> Bool {
+            switch self {
+            case .frame(let frame):
+                return frame.contains(point)
+            case .path(let path):
+                return path.contains(point)
+            }
+        }
+
+    }
+
+    // MARK: - Public Properties
+
+    /// The label of the accessibility element similar to a description read by VoiceOver when the element is brought into
+    /// focus.
+    public var label: String?
+
+    /// The value of the accessibility element similar to a description read by VoiceOver when the element is brought into
+    /// focus.
+    public var value: String?
+
+    /// The type of the accessibility element similar to a description read by VoiceOver when the element is brought into
+    /// focus.
+    public var type: UIAccessibilityTraits
+
+    /// The labels that will be used for user input.
+    public var userInputLabels: [String]?
+
+    /// The shape that will be highlighted on screen while the element is in focus.
+    public var shape: Shape
+
+    /// The object representing the accessibility node.
+    public var object: NSObject
```

Comment:
Looks like these properties are not meant to change after initialization. If so, it makes more sense to use let instead of var.